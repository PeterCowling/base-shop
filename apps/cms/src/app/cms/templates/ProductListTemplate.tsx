import React from "react";

export interface ProductListTemplateProps {
  /** Section heading */
  heading: string;
  /** Products to render in an unordered list */
  products: Array<{ id: string; name: string }>;
}

/**
 * Renders a basic list of products with a heading.
 */
export function ProductListTemplate({ heading, products }: ProductListTemplateProps) {
  return (
    <section>
      <h2>{heading}</h2>
      <ul>
        {products.map((p) => (
          <li key={p.id}>{p.name}</li>
        ))}
      </ul>
    </section>
  );
}
