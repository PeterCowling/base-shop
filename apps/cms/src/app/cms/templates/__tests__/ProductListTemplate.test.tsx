import { render, screen } from "@testing-library/react";
import { ProductListTemplate } from "../ProductListTemplate";

describe("ProductListTemplate", () => {
  it("renders list of products", () => {
    const products = [
      { id: "1", name: "One" },
      { id: "2", name: "Two" },
    ];
    const { container } = render(
      <ProductListTemplate heading="Featured" products={products} />
    );

    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
      "Featured"
    );
    expect(screen.getAllByRole("listitem")).toHaveLength(products.length);
    expect(container).toMatchSnapshot();
  });
});
