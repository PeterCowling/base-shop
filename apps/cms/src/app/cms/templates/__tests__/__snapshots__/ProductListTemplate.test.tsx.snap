// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListTemplate renders list of products 1`] = `
<div>
  <section>
    <h2>
      Featured
    </h2>
    <ul>
      <li>
        One
      </li>
      <li>
        Two
      </li>
    </ul>
  </section>
</div>
`;
